R/mrgsolve.R

```diff
@@ -18,6 +18,34 @@
 
 # @include mrgindata.R
 
+#' Check ETA names in the data set
+#' 
+#' This function checks to see if there are both `ETAn` and `ETn` in the data
+#' set when `n` is greater than 9. For example, when both `ETA12` and `ET12` 
+#' are found, an error is generated. This funciton is only use for its 
+#' side-effects. 
+#' @noRd
+check_etasrc_names <- function(data_names, neta, etasrc) {
+  eta_check <- paste0("ETA", seq(10, neta))
+  et_check <- paste0("ET", seq(10, neta))
```

Comment:
Mostly thinking aloud as reading:`dataobject::get_etas` only calls this when it's processing `i==10`, so the current code is never going to hit the `neta` less than 10 descending case (e.g., `seq(10,9) -> 10 9`). And even if somehow the calling code passed through `neta < 10`, it would just lead to wasted computation, not an error.

So, the current level of guarding seems sufficient for an internal function.

Reply:
Yes .... so the intention is that this _should_ only get called if there are more than 9 ETAs. So I think it should hold up as long as the behavior stays the same. 

What do you think about 

- generating an error in case `neta` <= 9 just for my own benefit in case the behavior changes?
- returning early in case `neta` is <= 9? just to reiterate that we don't need  this check in this case

Sort of opposites ... but wouldn't be against doing something like that just to protect me from my (future) self.

Reply:
I think adding a guard that errors for the `neta <= 9` case is a good idea: prevents future callers from making mistakes, as you mention, and also makes the intention clearer to the reader.